internal/pkg/composable/providers/kubernetes/hints_test.go

```diff
@@ -467,9 +466,128 @@ func TestGenerateHintsMappingWithProcessors(t *testing.T) {
 
 	hintData := GetHintsMapping(mapping, logger, "co.elastic", "asdfghjkl")
 
-	assert.Equal(t, expected_hints, hintData.composableMapping)
+	assert.Equal(t, expectedhints, hintData.composableMapping)
 	//assert.Equal(t, expected_procesors, hintData.processors). We replace this assertion with assert.Contains in order to avoid flakiness in tests because map keys are not sorted
-	assert.Contains(t, expected_procesors, hintData.processors[0])
-	assert.Contains(t, expected_procesors, hintData.processors[1])
+	if len(hintData.processors) > 0 {
+		assert.Contains(t, expectedprocesors, hintData.processors[0])
+		assert.Contains(t, expectedprocesors, hintData.processors[1])
+	}
+}
+
+// This test evaluates the hints Generation when you define specific container nginx
+// Following will need to include all annotations after top level "co.elastic.hints/" plus those that defined for nginx with prefix "co.elastic.hints.nginx"
+// mappings.container.name = nginx defines the container we want to emmit the new configuration. Annotations for other containers like co.elastic.hints.webapp should be excluded
+func TestGenerateHintsMappingWithProcessorsForContainer(t *testing.T) {
+	logger := getLogger()
+	// pod := &kubernetes.Pod{
```

Comment:
@constanca-m I commented out the pod section! Just kept it for reference with above tests.

Test is successful

Let me know if is ok or you think we can remove it?

Reply:
It's ok with me